bin/cli.js

```diff
@@ -29,10 +30,11 @@ const fsPromised = {
   writeFile: promisify(fs.writeFile),
 };
 
-function initKnex(env, opts) {
+const initKnex = async (env, opts) => {
   if (opts.esm) {
-    // enable esm interop via 'esm' module
-    require = require('esm')(module);
+    console.warn(
+      `The 'esm' opt is deprecated. Knex supports esm module imports using 'import()'`
+    );
```

Comment:
What happens if an end-user was already using the `--esm` flag?  Are we sure that the new functionality will be 100% compatible with their existing Knexfiles?  (In other words: will anything break for these users?)